src/components/tx/ApprovalEditor/hooks/useApprovalInfos.test.ts

```diff
@@ -95,6 +97,219 @@ describe('useApprovalInfos', () => {
+      amount: getTokenInfo.UNLIMITED_PERMIT2_AMOUNT.toBigInt(),
+      amountFormatted: PSEUDO_APPROVAL_VALUES.UNLIMITED,
+      spender: spenderAddress,
+      tokenAddress: '0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174'.toLowerCase(),
+      tokenInfo: undefined,
+      method: 'Permit2',
+    }
+
+    await waitFor(() => {
+      expect(result.current).toEqual([[mockApproval], undefined, false])
+    })
+  })
+
+  it('returns multiple ApprovalInfos for Permit2 PermitBatch message', async () => {
+    const spenderAddress = faker.finance.ethereumAddress()
+    const token1 = faker.finance.ethereumAddress()
+    const token2 = faker.finance.ethereumAddress()
+
+    const mockMessage: EIP712TypedData = {
+      types: {
+        EIP712Domain: [
+          {
+            name: 'name',
+            type: 'string',
+          },
+          {
+            name: 'chainId',
+            type: 'uint256',
+          },
+          {
+            name: 'verifyingContract',
+            type: 'address',
+          },
+        ],
+        PermitBatch: [
+          {
+            name: 'details',
+            type: 'PermitDetails[]',
+          },
+          {
+            name: 'spender',
+            type: 'address',
+          },
+          {
+            name: 'sigDeadline',
+            type: 'uint256',
+          },
+        ],
+        PermitDetails: [
+          {
+            name: 'token',
+            type: 'address',
+          },
+          {
+            name: 'amount',
+            type: 'uint160',
+          },
+          {
+            name: 'expiration',
+            type: 'uint48',
+          },
+          {
+            name: 'nonce',
+            type: 'uint48',
+          },
+        ],
+      },
+      domain: {
+        name: 'Permit2',
+        chainId: 137,
+        verifyingContract: '0x000000000022D473030F116dDEE9F6B43aC78BA3',
+      },
+      message: {
+        spender: spenderAddress,
+        sigDeadline: {
+          type: 'BigNumber',
+          hex: '0xffffffffffff',
+        },
+        details: [
+          {
+            token: token1,
+            amount: {
+              type: 'BigNumber',
+              hex: '0xffffffffffffffffffffffffffffffffffffffff',
+            },
+            expiration: {
+              type: 'BigNumber',
+              hex: '0xffffffffffff',
+            },
+            nonce: 0,
+          },
+          {
+            token: token2,
+            amount: {
+              type: 'BigNumber',
+              hex: '0xffffffffffffffffffffffffffffffffffffffff',
+            },
+            expiration: {
+              type: 'BigNumber',
+              hex: '0xffffffffffff',
+            },
+            nonce: 0,
+          },
+        ],
+      },
+    }
+
+    const { result } = renderHook(() => useApprovalInfos({ safeMessage: mockMessage }))
+
+    const expectedApprovals: ApprovalInfo[] = [
+      {
+        amount: getTokenInfo.UNLIMITED_PERMIT2_AMOUNT.toBigInt(),
```

Comment:
everywhere else i see Bignumber used, but here we do toBigInt(). Why?

Reply:
Good point. I think I had too many context switched to ethers v6. Replaced it with BigNumber!